### PR TITLE
Docs Reorg and Intro Page

### PR DESCRIPTION
**Pull request by aliabd**
- Adds an intro page to the docs, and moves python client, js client, and js components to their own section. 
- Removes the guides gallery page and instead puts it at the bottom of /docs 
- Redirects all old links 

Screen Shot 2024-04-18 at 10 22 19 AM

Design feedback welcome

Closes: #6630 
Closes: #5212

### CODE REVIEW

**Comment by freddyaboulton** [comment]
This is a super nice PR @aliabd ! I have some thoughts here but I think this is good enough to merge as is.

I think description of python and js clients should mirror each other
image

What are your thoughts on applying the styles used for the guide gallery to the custom components gallery? 
image

I think the section headers should use a bolder font
image

**Comment by abidlabs** [comment]
Very nice reorg @aliabd! 

> What are your thoughts on applying the styles used for the guide gallery to the custom components gallery?

Agree with @freddyaboulton, I'm not a huge fan of the washed out gray color text

Otherwise styling looks good to me! Tested on mobile and desktop

I have some ideas for the wording of the sections, will add later

**Comment by whitphx** [comment]
I really love this new design!

> I'm not a huge fan of the washed out gray color text

I agree with it.
I usually refer to [WCAG 2.0 guideline](https://www.w3.org/TR/WCAG20/) and [its checker](https://webaim.org/resources/contrastchecker/) for color accessibility. 
It says the current contrast is not enough, and I think we can use this as one reference.
CleanShot 2024-04-18 at 14 28 33@2x

File: js/_website/src/routes/[[version]]/docs/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+		>
+			<a class="mt-auto mb-2 flex-grow" href="./docs/gradio/" target="_self">
+				<div class="text-lg">
+					<p class="">Gradio</p>
+				</div>
+
+				<code class="font-light text-xs mb-2 text-gray-500">gradio</code>
+				<div class="  relative pr-4 text-smd font-normal text-gray-400">
+					Build and share machine learning demos and web apps, in just a few
+					lines of Python.
```
Comment (abidlabs) [inline js/_website/src/routes/[[version]]/docs/+page.svelte:67]:
```suggestion
					Build and share machine learning demos and web applications using the core Gradio Python library.
```

File: js/_website/src/routes/[[version]]/docs/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+			target="_self"
+			class="shadow-alternate hover:scale-[1.02] group group relative flex flex-col overflow-hidden rounded-xl bg-gradient-to-r px-3 py-4 dark:border-gray-900 from-{data
+				.COLOR_SETS[1]}-50 hover:shadow-alternate to-white shadow-none transition-shadow dark:from-gray-850 dark:to-gray-950 dark:hover:from-gray-800"
+		>
+			<div class="relative mb-2 flex items-center text-lg">
+				<span>Python Client</span>
+			</div>
+			<code class="font-light text-xs mb-2 text-gray-500">gradio-client</code>
+			<div class="relative pr-4 text-smd font-normal text-gray-400">
+				Turn any Gradio app into an API.
```
Comment (abidlabs) [inline js/_website/src/routes/[[version]]/docs/+page.svelte:126]:
```suggestion
				A Python client that allows you to make programmatic requests to Gradio applications.
```

File: js/_website/src/routes/[[version]]/docs/+page.svelte [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+			target="_self"
+			class="shadow-alternate hover:scale-[1.02] group group relative flex flex-col overflow-hidden md:first:row-span-2 rounded-xl bg-gradient-to-r px-3 py-4 dark:border-gray-900 from-{data
+				.COLOR_SETS[0]}-50 hover:shadow-alternate to-white shadow-none transition-shadow dark:from-gray-850 dark:to-gray-950 dark:hover:from-gray-800"
+		>
+			<div class="relative mb-2 flex items-center text-lg">
+				<span>Javascript Client</span>
+			</div>
+			<code class="font-light text-xs mb-2 text-gray-500">@gradio/client</code>
+			<div class="relative pr-4 text-smd font-normal text-gray-400">
+				A Javascript (and Typescript) client to call Gradio APIs.
```
Comment (abidlabs) [inline js/_website/src/routes/[[version]]/docs/+page.svelte:141]:
```suggestion
				A JavaScript (TypeScript) client that allows you to make programmatic requests to Gradio applications from the browser or server-side.
```

File: js/_website/src/routes/[[version]]/docs/+page.svelte [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+				href="../guides/custom-components-in-five-minutes"
+				target="_self"
+				class="shadow-alternate hover:scale-[1.02] group group relative flex flex-col overflow-hidden md:first:row-span-2 rounded-xl bg-gradient-to-r px-3 py-4 dark:border-gray-900 from-{data
+					.COLOR_SETS[5]}-50 hover:shadow-alternate to-white shadow-none transition-shadow dark:from-gray-850 dark:to-gray-950 dark:hover:from-gray-800"
+			>
+				<div class="relative mb-2 flex items-center text-lg">
+					<span>Custom Components</span>
+				</div>
+				<div class="relative pr-4 text-smd font-normal text-gray-400">
+					Create, use and share your own custom components.
```
Comment (abidlabs) [inline js/_website/src/routes/[[version]]/docs/+page.svelte:169]:
```suggestion
					Create, use, and share your own custom components within a Gradio application.
```

File: js/_website/src/routes/[[version]]/docs/+page.svelte [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+			class="shadow-alternate hover:scale-[1.02] group group relative flex flex-col overflow-hidden md:first:row-span-2 rounded-xl bg-gradient-to-r px-3 py-4 dark:border-gray-900 from-{data
+				.COLOR_SETS[4]}-50 hover:shadow-alternate to-white shadow-none transition-shadow dark:from-gray-850 dark:to-gray-950 dark:hover:from-gray-800"
+		>
+			<div class="relative mb-2 flex items-center text-lg">
+				<span>Gradio Lite</span>
+			</div>
+			<code class="font-light text-xs mb-2 text-gray-500">@gradio/lite</code>
+
+			<div class="relative pr-4 text-smd font-normal text-gray-400">
+				Serverless Gradio running entirely in your browser.
```
Comment (abidlabs) [inline js/_website/src/routes/[[version]]/docs/+page.svelte:185]:
```suggestion
				Run Gradio's Python code serverless (entirely in your browser) using WebAssembly.
```

File: js/_website/src/routes/[[version]]/docs/+page.svelte [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+				target="_self"
+				class="shadow-alternate hover:scale-[1.02] group group relative flex flex-col overflow-hidden md:first:row-span-2 rounded-xl bg-gradient-to-r px-3 py-4 dark:border-gray-900 from-{data
+					.COLOR_SETS[3]}-50 hover:shadow-alternate to-white shadow-none transition-shadow dark:from-gray-850 dark:to-gray-950 dark:hover:from-gray-800"
+			>
+				<div class="relative mb-2 flex items-center text-lg">
+					<span>Javascript Components</span>
+				</div>
+				<div class="relative pr-4 text-smd font-normal text-gray-400">
+					Use Gradio's Javascript components in custom components or outside the
+					context of Gradio.
```
Comment (abidlabs) [inline js/_website/src/routes/[[version]]/docs/+page.svelte:156]:
```suggestion
					Use Gradio's UI components in standalone JavaScript applications outside of the Gradio environment.
```

**Review by abidlabs — APPROVED** [review]
Left some suggestions on the wording for each sections. Hopefully this will increase the visibility of the whole gradio ecosystem and make the connections between the different sections a bit clearer.

**Comment by aliabd** [comment]
Thanks everybody for the reviews! Will clean up styling

**Comment by aliabd** [comment]
I changed up the style @freddyaboulton @abidlabs @whitphx 
Screen Shot 2024-04-23 at 11 28 28 AM

Let me know if you have any feedback. I'm working on a fix to a weird build error in the meantime, and will merge after

**Comment by abidlabs** [comment]
Looks great @aliabd! One small design feedback (just noticed now, feel free to address later), but I noticed that the buttons linking to specific sections in `gradio` have very round borders, which clash a bit with the less round borders in each of the different libraries. I think we should make the borders of the small buttons less round as well.

image

**Comment by aliabd** [comment]
@abidlabs 
Screen Shot 2024-04-23 at 1 05 50 PM

**Pull request merged**